### PR TITLE
fix: [CDS-46816]: NG ServiceV2 - Render expressions in ACR Artifact Parameter Fields

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
This PR fixes the auto-loading of ACR artifact information in the "input variables" modal that appears when user manually runs a pipeline.

I created new endpoints for fetching subscription Ids, registries, repositories for ACR artifacts. These new endpoints receive the pipeline Id and input YAML data, which allows for expression rendering for the ACR parameters using variables declared in the pipeline. Also, I fixed the buildDetailsV2 API to render the `connectorRef` expression if necessary.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)

### CODE REVIEW

**Comment by vivekkarnatiharness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 111 at r1](https://reviewable.io/reviews/harness/harness-core/40978#-NIFsF766BDQvUhvfuV6:-NIFsF766BDQvUhvfuV7:bskg0ps) ([raw file](https://github.com/harness/harness-core/blob/0b9aec51253878671fb259924cc726557f1de64c/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L111)):*
> ```Java
>             pipelineIdentifier, runtimeInputYaml, acrArtifactConfig.getStringParameterFields(), fqnPath,
>             gitEntityBasicInfo, serviceRef);
>         azureConnectorIdentifier = acrArtifactConfig.getConnectorRef().getValue();
> ```

azureConnectorIdentifier = (String)acrArtifactConfig.getConnectorRef().fetchFinalValue();
fetchfinalvalue must be used,it handles case of expression



_Code quote:_
```Java
azureConnectorIdentifier = acrArtifactConfig.getConnectorRef().getValue();
```

**Comment by vivekkarnatiharness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 182 at r1](https://reviewable.io/reviews/harness/harness-core/40978#-NIFskXv2tHnP0EbGXHW:-NIFskXv2tHnP0EbGXHX:b-xv5qhm) ([raw file](https://github.com/harness/harness-core/blob/0b9aec51253878671fb259924cc726557f1de64c/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L182)):*
> azureConnectorIdentifier = (String)acrArtifactConfig.getConnectorRef().fetchFinalValue();
> fetchfinalvalue must be used,it handles case of expression

**Comment by vivekkarnatiharness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 185 at r1](https://reviewable.io/reviews/harness/harness-core/40978#-NIFstbS9b_NtWl9jO43:-NIFstbS9b_NtWl9jO44:b-xiucu8) ([raw file](https://github.com/harness/harness-core/blob/0b9aec51253878671fb259924cc726557f1de64c/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L185)):*
> ```Java
>       }
>       if (isEmpty(subscriptionId)) {
>         subscriptionId = acrArtifactConfig.getSubscriptionId().getValue();
> ```

Same here as well!

**Comment by vivekkarnatiharness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 182 at r1](https://reviewable.io/reviews/harness/harness-core/40978#-NIFskXv2tHnP0EbGXHW:-NIFujjF4Wnf9WYYDwtD:b62c8wh) ([raw file](https://github.com/harness/harness-core/blob/0b9aec51253878671fb259924cc726557f1de64c/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L182)):*
<details><summary><i>Previously, vivekkarnatiharness wrote…</i></summary><blockquote>

> azureConnectorIdentifier = (String)acrArtifactConfig.getConnectorRef().fetchFinalValue();
> fetchfinalvalue must be used,it handles case of expression

</blockquote></details>

Sorry,it will work ,as we are resolving the expression before assigning itself.

**Review by vivekkarnatiharness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: 3 file(s) to review. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 111 at r1](https://reviewable.io/reviews/harness/harness-core/40978#-NIFsF766BDQvUhvfuV6:-NIFtratAWSW0mTPotXF:b62c8wh) ([raw file](https://github.com/harness/harness-core/blob/0b9aec51253878671fb259924cc726557f1de64c/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L111)):*
<details><summary><i>Previously, vivekkarnatiharness wrote…</i></summary><blockquote>

azureConnectorIdentifier = (String)acrArtifactConfig.getConnectorRef().fetchFinalValue();
fetchfinalvalue must be used,it handles case of expression
</blockquote></details>

Sorry,it will work ,as we are resolving the expression before assigning itself.

**Comment by vcalasansh** [comment]
trigger smartchecks

**Review by shivamnegi94 — APPROVED** [review]

**Review by shivamnegi94 — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 2 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: 4 file(s) to review. 1 of 2 approvals obtained.

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java [hunk 1]
```diff
@@ -130,6 +159,38 @@ public ResponseDTO<AcrRegistriesDTO> getRegistriesBySubscription(
         acrResourceService.getRegistries(connectorRef, orgIdentifier, projectIdentifier, subscriptionId));
   }
 
+  @POST
+  @Path("v3/container-registries")
```
Comment (sribalij) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java:163]: Is the UI going to call all the new versons going forward?

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: Some of the checks are too old. 4 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 182 at r1](https://reviewable.io/reviews/harness/harness-core/40978#-NIFskXv2tHnP0EbGXHW:-NJ5anNu8R7WaQmZ5aNd:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b9aec51253878671fb259924cc726557f1de64c/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L182)):*
<details><summary><i>Previously, vivekkarnatiharness wrote…</i></summary><blockquote>

Sorry,it will work ,as we are resolving the expression before assigning itself.
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 163 at r4](https://reviewable.io/reviews/harness/harness-core/40978#-NIEsBMQIJXsshU7B9fQ-r4-163:-NJ5aeIlB7Gm_B47Ae0y:b1lqgrg) ([raw file](https://github.com/harness/harness-core/blob/095a2aa184de9d5fdb2043cfb202546e3c0efb15/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L163)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

Is the UI going to call all the new versons going forward?
</blockquote></details>

That's right, I will open a UI ticket for getting UI requests migrated to these new endpoints.

**Review by sribalij — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: Some of the checks are too old. 4 file(s) to review. 1. unresolved discussions. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 163 at r4](https://reviewable.io/reviews/harness/harness-core/40978#-NIEsBMQIJXsshU7B9fQ-r4-163:-NJ6qoye3aahhihKQWTU:b-4hvi0m) ([raw file](https://github.com/harness/harness-core/blob/095a2aa184de9d5fdb2043cfb202546e3c0efb15/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L163)):*
<details><summary><i>Previously, vcalasansh (Vinícius Calasans) wrote…</i></summary><blockquote>

That's right, I will open a UI ticket for getting UI requests migrated to these new endpoints.
</blockquote></details>

Got it.

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: Some of the checks are too old. 4 file(s) to review. 2. unresolved discussions. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java` line 163 at r4](https://reviewable.io/reviews/harness/harness-core/40978#-NIEsBMQIJXsshU7B9fQ-r4-163:-NJ79eyL0C8GTYLPO6dj:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/095a2aa184de9d5fdb2043cfb202546e3c0efb15/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/acr/AcrArtifactResource.java#L163)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

Got it.
</blockquote></details>

Done.

**Review by sribalij — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: 1 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by sribalij — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by shivamnegi94 — APPROVED** [review]

**Review by shivamnegi94 — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: 1. unresolved discussions.

**Review by vivekkarnatiharness — COMMENTED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: 1. unresolved discussions.

**Review by vivekkarnatiharness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: :shipit: complete! ✓

**Comment by vcalasansh** [comment]
trigger smartchecks

**Review by sribalij — APPROVED** [review]
> Reviewed 1 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by shivamnegi94 — APPROVED** [review]

**Review by shivamnegi94 — APPROVED** [review]
> Reviewed 1 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40978)* status: :shipit: complete! ✓

**Pull request merged**